### PR TITLE
Use the Jackson BOM to solve Jackson dependency errors

### PR DESCRIPTION
**Pull request by jdubois**
Using Azure OpenAI, on some Jackson mappings I'm getting the following error:
```
Caused by: java.lang.NoClassDefFoundError: com/fasterxml/jackson/annotation/JsonTypeInfo$Value
```
This is because not all our Jackson dependencies are using the same version. The intent of this commit is to fix this, by using the Jackson BOM.

### CODE REVIEW

File: langchain4j-parent/pom.xml [hunk 1]
```diff
@@ -266,6 +265,13 @@
...
 
+            <dependency>
+                <groupId>com.fasterxml.jackson</groupId>
+                <artifactId>jackson-bom</artifactId>
+                <version>${jackson.version}</version>
+                <type>pom</type>
+            </dependency>
+
             <dependency>
                 <groupId>com.fasterxml.jackson.core</groupId>
```
Comment (langchain4j) [inline langchain4j-parent/pom.xml]: Now once you've added `jackson-bom`, does it make sense to list all jackson dependencies here?

Comment (shalk) [reply langchain4j-parent/pom.xml]: also. if add bom in  `<dependencyManagement>`, i think  need add `<scope> import </scope> `

Comment (jdubois) [reply langchain4j-parent/pom.xml]: You are both right, I used the import scope and simplified the code

**Comment by shalk** [comment]
LGTM

**Review by ghost — APPROVED** [review]
@jdubois thank you!

**Pull request merged**